src/otaclient/ota_core.py

```diff
@@ -171,8 +171,8 @@ def __init__(
         cookies_json: str,
         ca_chains_store: CAChainStore,
         upper_otaproxy: str | None = None,
-        boot_controller: BootControllerProtocol,
-        create_standby_cls: type[RebuildMode],
+        boot_controller: BootControllerProtocol | None = None,
+        create_standby_cls: type[RebuildMode] | None = None,
```

Comment:
These variables are necessary only for data updates(`_OTAUpdater`) class, not used in this base class(and client update).
To keep Liskov, declare in base class and set as Optional.